tests/bson-serialize-cursor-id.phpt

```diff
@@ -0,0 +1,25 @@
+--TEST--
+MongoDB\Driver\CursorID serialization
+--FILE--
+<?php
+include dirname(__FILE__) . '/utils.inc';
+
+$m = new MongoDB\Driver\Manager("mongodb://localhost:27017");
+cleanup( $m );
+
+$bw = new MongoDB\Driver\BulkWrite( [ 'ordered' => false ] );
+$bw->insert( [ 'test' => 'foo', 'i' => 1 ] );
+$bw->insert( [ 'test' => 'foo', 'i' => 2 ] );
+$bw->insert( [ 'test' => 'foo', 'i' => 3 ] );
+$result = $m->executeBulkWrite( 'demo.test', $bw );
+
+$q = new MongoDB\Driver\Query( [], [ 'batchSize' => 2 ] );
+$cursorId = $m->executeQuery( 'demo.test', $q )->getId();
+
+echo "result:\n";
+hex_dump( MongoDB\BSON\fromPHP( [ 'cid' => $cursorId ] ) );
+?>
+--EXPECTF--
+result:
+     0 : 12 00 00 00 12 63 69 64 00 %s %s %s %s %s %s %s  [.....cid.%s]
+    10 : 00 00                                            [..]
```

Comment:
You're only matching 7 bytes with `%s %s %s %s %s %s %s` and may be assuming the last byte of the cursor ID is `00`. I'd suggest replacing all of this with eight `%x` patterns seprated by spaces. The `%s` in the brackets to the right can stay as-is, but you should replace `[..]` on the second line with `[%c.]`.